### PR TITLE
Add the repro case for race condition in revisit_written_pages as a GC test case

### PR DESCRIPTION
**Pull request by PeterSolMS**
This was github issue #68443 (https://github.com/dotnet/runtime/issues/68443).

### CODE REVIEW

**Comment by mangod9** [comment]
Does it also need to be added to a CI config, or is the expectation that this is run manually?

**Comment by Maoni0** [comment]
I think Peter ran this standalone manually in a loop. I'm fine with keeping this in the stress dir even though this is not added to the testmix config as we will be doing work on stress in the near future anyway (I suspect lots of tests will be removed/changed/etc with new tests added).

**Review by Maoni0 — APPROVED** [review]

File: src/tests/GC/Stress/Tests/DictionaryGrowth.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            ThreadStart ts = new ThreadStart(Run);
+            Thread t = new Thread(ts);
+            t.Start();
+            ta[i] = t;
+        }
+        for (int i = 0; i < threadCount; i++)
+        {
+            ta[i].Join();
+        }
+        double elapsed = 0.001*(System.Environment.TickCount - startTick);
```
Comment (danmoseley) [inline src/tests/GC/Stress/Tests/DictionaryGrowth.cs:41]: is 0.001 right? There are 10 million ticks in a second

Comment (Maoni0) [reply src/tests/GC/Stress/Tests/DictionaryGrowth.cs:41]: Environment.TickCount returns milliseconds.

**Pull request merged**